includes/tracks/class-wc-tracks-client.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/**
+ * Send Tracks events on behalf of a user.
+ *
+ * @class   WC_Tracks_Client
+ * @package WooCommerce/Classes
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Class WC_Tracks_Client.
+ */
+class WC_Tracks_Client {
+	const PIXEL           = 'https://pixel.wp.com/t.gif';
+	const BROWSER_TYPE    = 'php-agent';
+	const USER_AGENT_SLUG = 'tracks-client';
+
+	/**
+	 * Record a Tracks event
+	 *
+	 * @param  array $event Array of event properties
+	 * @return bool|WP_Error         True on success, WP_Error on failure.
+	 */
+	public static function record_event( $event ) {
+		if ( ! $event instanceof WC_Tracks_Event ) {
+			$event = new WC_Tracks_Event( $event );
+		}
+		if ( is_wp_error( $event ) ) {
+			return $event;
+		}
+
+		$pixel = $event->build_pixel_url( $event );
+
+		if ( ! $pixel ) {
+			return new WP_Error( 'invalid_pixel', 'cannot generate tracks pixel for given input', 400 );
+		}
+
+		return self::record_pixel( $pixel );
+	}
+
+	/**
+	 * Synchronously request the pixel.
+	 *
+	 * @param string $pixel pixel url and query string.
+	 * @return bool|WP_Error         True on success, WP_Error on failure.
+	 */
+	public static function record_pixel( $pixel ) {
+		// Add the Request Timestamp and URL terminator just before the HTTP request.
+		$pixel .= '&_rt=' . self::build_timestamp() . '&_=_';
+
+		$response = wp_remote_get(
+			$pixel,
+			array(
+				'blocking'    => true, // The default, but being explicit here.
+				'timeout'     => 1,
+				'redirection' => 2,
+				'httpversion' => '1.1',
```

Comment:
r/e redirection and timeout, why specifically these values? Is timeout of 1 enough? If it does timeout, does this retry, and could that lead to performance issues if the host was unreachable for a period of time?

Reply:
I removed `timeout` in favour of the default (5). These values are used by the Jetpack implementation and I can't find any reasoning behind them. The timeout seems short, but I don't know why redirection would be used, so limiting it makes sense.